bliss/simulator/prior.py

```diff
@@ -228,12 +249,14 @@ def _load_color_distribution(self, sdss_fields):
 
                 for obj, _ in enumerate(objc_type):
                     if objc_type[obj] == 6 and thing_id[obj] != -1 and torch.all(fluxes[obj] > 0):
-                        # flux relative to r-band
-                        ratios = fluxes[obj] / fluxes[obj][2]  # noqa: WPS220
+                        ratios = self._compute_flux_ratio(  # noqa: WPS220
+                            fluxes[obj], run, camcol, field, field_dir, gains
+                        )
```

Comment:
This can be moved out of the conditional since it's the same in each branch

Reply:
This was tricky because I suppose we don't want to compute the ratios for objects that aren't 'things' in the catalog. It's not clean at all really but I think we can just combine the logic in both conditions [i.e. if (the object is a star OR a galaxy) AND `(thing_id[obj] != -1)` AND `torch.all(fluxes[obj] > 0)`].

Reply:
Oh, I see. I guess that makes sense. Though another problem is that `_compute_flux_ratio` is called for each source in the for loop, and it's loads the fits files for each band each time - that's really inefficient. I'm more convinced that just passing an SDSS object is the way to go, ensuring everything is only loaded once. And that will make it easier to integrate with the changes in #853, which ideally will make it so this object gets passed down from the top.